sdks/python/apache_beam/runners/worker/sdk_worker.py

```diff
@@ -628,8 +632,13 @@ def _blocking_request(self, request):
       return response
 
   def _next_id(self):
-    self._last_id += 1
-    return str(self._last_id)
+    with self._lock:
+      # Use a lock here because this GrpcStateHandler is shared across all
+      # requests which have the same process bundle descriptor. State requests
+      # can concurrently access this section if a Runner uses threads / workers
+      # (aka "parallelism") to send data to this SdkHarness and its workers.
+      self._last_id += 1
```

Comment:
Could you assign the result to a local variable to avoid the lock?

Reply:
You mean avoiding the lock during `str(id)`? like so:

```python
with self._lock:
  self._last_id += 1
  request_id = self._last_id
return str(request_id)
```

Could be a potential performance difference.